ams/AzureResourceProvider.cs

```diff
@@ -15,6 +16,8 @@ class AzureResourceProvider
         protected readonly GlobalOptions _globalOptions;
         protected readonly TokenCredential _credentials;
 
+        public Dictionary<string, ResourceGroupResource> resourceGroups { get; set; }
```

Comment:
It is recommended to use a name starting with an upper case letter for a public property.

resourceGroups  to  ResourceGroups.

To be more specific here,  the property name might be StorageResourceGroups,  since it is for Storage accounts only.